File: TC_login.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
import time


url = "https://www.saucedemo.com/"
driver = webdriver.Chrome()
wait = WebDriverWait(driver, 5) 

def login(username, password):
    driver.get(url)
    driver.find_element(By.ID, "user-name").clear()
    driver.find_element(By.ID, "user-name").send_keys(username)
    driver.find_element(By.ID, "password").clear()
    driver.find_element(By.ID, "password").send_keys(password)
    driver.find_element(By.ID, "login-button").click()

def get_error_message():
    try:
        error_element = wait.until(EC.visibility_of_element_located((By.XPATH, "//h3[@data-test='error']")))
        return error_element.text.strip()
    except:
        return None

print("Test Case 1: Login dengan kredensial benar")
login("standard_user", "secret_sauce")
if "inventory" in driver.current_url:
    print("PASS: Login berhasil")
else:
    print("FAIL: Login gagal")
time.sleep(1)

print("\nTest Case 2: Login dengan password salah")
login("standard_user", "wrong_password")
error_message = get_error_message()
print(f"[DEBUG] Pesan error: {error_message}")
if error_message and "do not match any user" in error_message.lower():
    print("PASS: Error message muncul")
else:
    print("FAIL: Tidak ada atau salah error message")
time.sleep(1)

def get_error_message():
    try:
        error_element = wait.until(
            EC.visibility_of_element_located((By.XPATH, "//h3[@data-test='error']"))
        )
        return error_element.text.strip()
    except Exception as e:
        print(f"[DEBUG] Tidak ada error message: {e}")
        return None

print("\nTest Case 3: Login tanpa username")
login("", "secret_sauce")

time.sleep(0.5)

error_message = get_error_message()
print(f"[DEBUG] Pesan error: {error_message}")
if error_message and "username is required" in error_message.lower():
    print("PASS: Error message muncul")
else:
    print("FAIL: Tidak ada atau salah error message")

driver.quit()

File: test_javascript/TC_Login.js
const { Builder, By, until } = require('selenium-webdriver');

(async function saucedemoTests() {
    const driver = await new Builder().forBrowser('chrome').build();
    const waitTime = 5000;
    const url = 'https://www.saucedemo.com/';

    async function login(username, password) {
        await driver.get(url);
        const userField = await driver.findElement(By.id('user-name'));
        const passField = await driver.findElement(By.id('password'));
        const loginBtn = await driver.findElement(By.id('login-button'));

        await userField.clear();
        await userField.sendKeys(username);
        await passField.clear();
        await passField.sendKeys(password);
        await loginBtn.click();
    }

    async function getErrorMessage() {
        try {
            const errorElement = await driver.wait(
                until.elementLocated(By.xpath("//h3[@data-test='error']")),
                waitTime
            );
            return (await errorElement.getText()).trim();
        } catch (e) {
            return null;
        }
    }

    // Test Case 1: Login dengan kredensial benar
    console.log('Test Case 1: Login dengan kredensial benar');
    await login('standard_user', 'secret_sauce');
    if ((await driver.getCurrentUrl()).includes('inventory')) {
        console.log('PASS: Login berhasil');
    } else {
        console.log('FAIL: Login gagal');
    }

    // Test Case 2: Login dengan password salah
    console.log('\nTest Case 2: Login dengan password salah');
    await login('standard_user', 'wrong_password');
    let errorMessage = await getErrorMessage();
    console.log('[DEBUG] Pesan error:', errorMessage);
    if (errorMessage && errorMessage.toLowerCase().includes('do not match any user')) {
        console.log('PASS: Error message muncul');
    } else {
        console.log('FAIL: Tidak ada atau salah error message');
    }

    // Test Case 3: Login tanpa username
    console.log('\nTest Case 3: Login tanpa username');
    await login('', 'secret_sauce');
    errorMessage = await getErrorMessage();
    console.log('[DEBUG] Pesan error:', errorMessage);
    if (errorMessage && errorMessage.toLowerCase().includes('username is required')) {
        console.log('PASS: Error message muncul');
    } else {
        console.log('FAIL: Tidak ada atau salah error message');
    }

    await driver.quit();
})();